packages/legend-application-data-cube/src/stores/LegendDataCubeDataCubeCacheEngine.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as duckdb from '@duckdb/duckdb-wasm';
+import duckdb_wasm from '@duckdb/duckdb-wasm/dist/duckdb-mvp.wasm';
+import duckdb_wasm_next from '@duckdb/duckdb-wasm/dist/duckdb-eh.wasm';
+import {
+  TDSExecutionResult,
+  TDSRow,
+  TabularDataSet,
+} from '@finos/legend-graph';
+import { action, makeObservable, observable } from 'mobx';
+import type { AsyncDuckDBConnection } from '@duckdb/duckdb-wasm';
+
+export class LegendDataCubeDataCubeCacheEngine {
```

Comment:
```suggestion
export class LegendDataCubeCacheManager {
```

Reply:
Same thing here, let's move this inside `LegendDataCubeDataCubeEngine`. There's no real reason (for now) to have this separate.